enhancements/monitoring/monitoring-config.md

```diff
@@ -0,0 +1,307 @@
+        Add controller-gen logic to makefile
+        Add API to api/config/v1
+        Add Generated CRD: config/v1/zz_generated.crd-manifests/0000_10_config-operator_01_clustermonitoring.crd.yaml
+        Add example CustomResource: config/examples/clustermonitoringoperator.yaml
+    - Client codegen 
+    - Reconcile logic: https://github.com/openshift/cluster-monitoring-operator/pull/2350
+    - Add decoupling ConfigMap / CustomResource:
+        Controller logic is strongly dependant of *manifests.Config struct.
+        
+
+### Example configuration
+
+
+#### current ConfigMap
+
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    telemeterClient:
+      enabled: false
+    prometheusK8s:
+      volumeClaimTemplate:
+        metadata:
+          name: prometheus-data
+          annotations:
+            openshift.io/cluster-monitoring-drop-pvc: "yes"
+        spec:
+          resources:
+            requests:
+              storage: 20Gi
+```
+
+### CRD
+
+```
+apiVersion: clustermonitoring.config.openshift.io
+kind: ClusterMonitoring
+metadata:
+  name: cluster
+  namespace: openshift-monitoring
+spec:
+  prometheusK8s:
+    volumeClaimTemplate:
+      metadata:
+        name: prometheus-data
+        annotations:
+          openshift.io/cluster-monitoring-drop-pvc: "yes"
+      spec:
+        resources:
+          requests:
+            storage: 20G
+```
+
+
+### Workflow Description
+
+See example in the [Proposal](#Proposal) for details.
+
+### API Extensions
+
+Add two new CRD's:
+- ClusterMonitoring
+- UserWorkloadClusterMonitoring
+
+### Topology Considerations
+
+Will function identically in all topologies.
+
+#### Hypershift / Hosted Control Planes
+No special considerations.
+
+#### Standalone Clusters
+No special considerations.
+
+#### Single-node Deployments or MicroShift
+No special considerations.
+
+### Implementation Details/Notes/Constraints
+
+
+### Migration path
+
+Migration path it will be split in three phases
+
+- Phase 1: Pre-GA (Gradual Migration for Developers)
+  - In this phase only developers are expected to use CRD.
+  - Both ConfigMap and CRD will coexist for a while until the transition is complete.
+  - Work with a Feature flag.
+  - Two options: ConfigMap or CRD.
+      Preference is to use ConfigMap over CRD during this phase.
+  - Behavior: If an entry exists in the ConfigMap, ignore the CRD.
+      If there’s no entry in the ConfigMap, use the CRD.
+  - Merge Process (only for phase 1)
+      Top-level fields (e.g., AlertmanagerMainConfig *AlertmanagerMainConfig || spec.AlertmanagerMainConfig):
+      If a field is nil in the ConfigMap, use the CRD; otherwise, stick with the ConfigMap.
+
+- Phase 2: Transition (User Control)
+  Discuss whether to keep dual support for a couple of releases.
+  - New Logic:
+    Allow users to choose the source of configuration (ConfigMap or CRD, warn about CRD preference).
```

Comment:
By having a config somewhere, or just the presence?

Reply:
We can warn the user it is possible to use both, but always CR preference, but next point is to add some logic to do it.

Reply:
Can you explain in the doc how you'll do that? This should be represented in the workflows. A workflow that shows what a user would see when they apply the CR or configmap in this dual state would really help

Reply:
I am not sure to understand it. If user choose a CR, logic take CR and monitoring stack is configured. If user choose ConfigMap, then there is no CR and logic take ConfigMap. If the is the case we have both, we use CR preference.

Reply:
So if the user creates the CR, then a different user on the same cluster updates the configmap, expecting the change to take effect, they won't see any change since it will be overridden by the CR.

How would that user realise that they are configuring the wrong object? Can we build something (ValidatingAdmissionPolicy) that would warn them if they update the configmap when the CR exists?

Reply:
Yep it can be a hook. We can choose the best option to warn the user.

Reply:
I don't mean a hook, I mean ValidatingAdmissionPolicy specifically, it's more performant and easier to write than a webhook

Reply:
Add new info, please close discussion or add suggestion in order to clean the PR. Thanks!